### DISCUSSION TITLE
How to switch "WHERE" operators?

### DISCUSSION THREAD

**Discussion by amneher** [sqlalchemy/sqlalchemy#9299] [Usage Questions]
In sqlalchemy/lib/orm/evaluator.py, in the EvaluatorCompiler, the process() method does this:
```py
class EvaluatorCompiler:
    def __init__(self, target_cls=None):
        self.target_cls = target_cls

    def process(self, clause, *clauses):
        if clauses:
            clause = **and_(clause, *clauses)**`
```
Is there a way to switch to or_ ?

**Comment by zzzeek** [comment]
hi -

this is a really obscure use inside of an ORM feature that is used in a very limited way (and only optionally).    The change you describe has no rationale and there is likely a much better way to solve the problem that you're having, so share what the actual issue is, thanks!

**Comment by amneher** [comment]
I have queries with several user-accessible fields, currently they're all-or-nothing since WHERE clauses are constructed with AND. I'd like users to be able to get records that match or more of the criteria.

**Reply by zzzeek** [reply]
the evaluator you refer towards has nothing to do with that function...and as I type this, I realize, are you using the "orm/evaluator.py" module directly?    That's a private module that's not supported for direct use.   it's not documented at all, I will add a note that it's for private use.   if you like what it does, I would copy it and just use your own version of it.

**Reply by zzzeek** [reply]
I will be pushing a change to rename this to `_EvaluatorCompiler`, I will leave the `EvaluatorCompiler` name in place but will be removed in a future major release.

**Reply by amneher** [reply]
Thanks for the info, I'll leave that class alone.

**Reply by zzzeek** [reply]
feel free to copy it and use your local copy in any way you want!   I just can't maintain the one that's shipped in the release for people trying to do new things with it.
